metadata/src/lib.rs

```diff
@@ -98,13 +98,7 @@ fn get_type_def_hash(
         }
         TypeDef::Variant(variant) => {
             let mut bytes = Vec::new();
-            // The type at path `node_template_runtime::Call` contains variants of the pallets
-            // registered in order. Swapping the order between two pallets would result
-            // in a different hash, but the functionality is still identical.
-            // Sort by variant name to result in deterministic hashing.
-            let mut variants: Vec<_> = variant.variants().iter().collect();
-            variants.sort_by_key(|variant| variant.name());
-            for var in variants {
+            for var in variant.variants().iter() {
```

Comment:
I removed this logic; it seems like we sort pallets (or should sort them) at the level of getting an entire-metadata hash, which I think will satisfy why we are sorting them?

Also, we can't sort _any_ variant type in this way, because variant types will be SCALE encoded/decoded taking into account the variant index, and so if some of the variants move around normally, it's a breaking change. We can _only_ do this at th pallet level as a special one-off to allow pallets to move around (and probably at the call level as well because our runtime code allows these to be shuffled around too I think, but would need to check!)

Removing this didn't seem to break any tests, but def worth making sure we aren't breaking any use cases and don't need to have this sort of logic somewhere else!